src/quic/sessionticket.cc

```diff
@@ -0,0 +1,177 @@
+#if HAVE_OPENSSL && NODE_OPENSSL_HAS_QUIC
+
+#include "sessionticket.h"
+#include <env-inl.h>
+#include <memory_tracker-inl.h>
+#include <ngtcp2/ngtcp2_crypto.h>
+#include <node_buffer.h>
+#include <node_errors.h>
+
+namespace node {
+
+using v8::ArrayBufferView;
+using v8::Just;
+using v8::Local;
+using v8::Maybe;
+using v8::MaybeLocal;
+using v8::Nothing;
+using v8::Object;
+using v8::Value;
+using v8::ValueDeserializer;
+using v8::ValueSerializer;
+
+namespace quic {
+
+namespace {
+std::optional<SessionTicket::AppData::Source*> GetAppDataSource(SSL* ssl) {
```

Comment:
FWIW, I am not a huge fan of using `std::optional<T*>` unless a non-empty optional holding a `(T*) nullptr` is a valid value. I personally don't see much of a benefit compared to returning a plain pointer, which is already a nullable reference. (Then again, we do often use pointers where other modern C++ projects would use references, so...)